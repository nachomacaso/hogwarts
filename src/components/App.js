import React, { Component } from 'react';
import { BrowserRouter as Router, Route, Redirect } from "react-router-dom";
import Header from './Header';
import StudentIndex from './StudentIndex';
import StudentEdit from './StudentEdit';
import StudentCreate from './StudentCreate';
import '../styles/index.css';

class App extends Component {
  render() {
    return (
      <div className="App">
        <Header/>
        <Router>
          <div>
            <Route exact path="/" render={() => (<Redirect to="/students" />)}/>
            <Route exact path="/students" component={StudentIndex} />
            <Route exact path="/students/create" component={StudentCreate} />
            <Route exact path="/students/:id/edit" component={StudentEdit} />
          </div>
        </Router>
      </div>
    );
  }
}

export default App;
