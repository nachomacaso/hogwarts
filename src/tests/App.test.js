import React from 'react';
import { shallow } from 'enzyme';
import App from '../components/App';

describe('App', () => {
  it('should render correctly', () => {
    const component = shallow(<App />);
  
    expect(component).toMatchSnapshot();
  });
});