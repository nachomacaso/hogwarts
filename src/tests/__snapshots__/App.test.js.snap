// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<div
  className="App"
>
  <WithStyles(Header) />
  <BrowserRouter>
    <div>
      <Route
        exact={true}
        path="/"
        render={[Function]}
      />
      <Route
        component={[Function]}
        exact={true}
        path="/students"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/students/create"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/students/:id/edit"
      />
    </div>
  </BrowserRouter>
</div>
`;
